File: dataset/avatar/Python/Code/codeforces_500_A.py
import sys
n , t = map ( int , input ( ).split ( ) )
s = list ( [ int ( x ) for x in input ( ).split ( ) ] )
current_cell = 0
while current_cell <= t :
    current_cell += s [ current_cell - 1 ]
    if current_cell == t :
        print ( "YES" )
        sys.exit ( )
    else :
        continue
print ( "NO" )

File: output/qwen_2.5_72b_instruct/avatar/algo_based_translation/python/java/codeforces_500_A.java

import java.util.Scanner;
import java.util.List;
import java.util.ArrayList;

public class codeforces_500_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int t = scanner.nextInt();
        List<Integer> s = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            s.add(scanner.nextInt());
        }
        int currentCell = 0;
        while (currentCell <= t) {
            currentCell += s.get(currentCell - 1);
            if (currentCell == t) {
                System.out.println("YES");
                System.exit(0);
            }
        }
        System.out.println("NO");
    }
}

